Tutorials/Creating-and-distributing-a-package/index.md

```diff
@@ -0,0 +1,500 @@
+
+For now we don't need to worry about the output directory option, we will let UmbPack save it in the current folder. Likewise with the name and version override, we will let UmbPack use the name and version we specified in the package.xml file. For the package.xml path we will specify the one in the root we had a look at right before this, that points to both the dll file and the App_Plugins folder in the website project.
+
+```umbpack pack .\package.xml```
+
+And now we have a zipped version of our package with any new additions in App_Plugins automatically added and listed in the package.xml of the zip. Now you can make changes and run the above command to have an updated version of the package within seconds!
+
+Next up - let's push this package update to Our from the commandline!
+
+## Pushing your package to Our using UmbPack
+
+To push a package update to Our with UmbPack we need to use the `push` command, so let's have a quick look at the options for that by running:
+
+```umbpack push -h```
+
+You will see there are 2 nessecary options, an API key and a path to the package zip. Then it is also possible for you to specify the Umbraco versions and dotnet version the package is compatible with, and also to archive old packages and set this as the new current package.
+
+In short - similar options to what you can set on the package upload section on Our when uploading a package:
+
+![Our package data][our-pkg-data]
+
+So before we can try this out, let's go to Our and create an API key!
+
+### Creating an Our API key
+
+If you had back to Our Umbraco and visit the [package overview][member-package-overview], then you will notice that there is a button under your package that says "API Keys". 
+Each package api key is tied to that specific package, if you visit the page then you can create keys with a name you can use to differentiate the keys. Once you make a key it will show once, as soon as you refresh or navigate away the key will be gone and you'll have to make a new one if you lose it.
+
+Once you created your key, make sure to copy and paste it somewhere. We will need to use it a few times in the coming steps, remember if you lose it you will have to make a new one!
+
+### Pushing to Our with UmbPack
+
+Now that we have an API key we can try to push our package update to Our Umbraco. This can be done like this:
+
+```umbpack push .\PackageWorkshopDashboard_1.0.0.zip -k [Api key here]```
+
+:::note 
+If a package with the same name already exists it may give an error. In that case you can run `umbpack pack .\package.xml -v 1.0.1` to create a new version of the package then push it with the above command after editing the path to the new package.
+:::
+
+An important thing to note with the push command here is that it sets some defalt values. If not specified otherwise it will default to saying your package is compatible with Umbraco v8.5.0, Dotnet 4.7.2 and it's to be set as the new "current" package on Our.
```

Comment:
```suggestion
An important thing to note with the push command here is that it sets some default values. If these values are not explicitly set it will default to saying your package is compatible with Umbraco v8.5.0, Dotnet 4.7.2 and it's to be set as the new "current" package on Our.
```